scene/2d/collision_shape_2d.cpp

```diff
@@ -39,14 +39,37 @@ void CollisionShape2D::_shape_changed() {
 	queue_redraw();
 }
 
+CollisionObject2D *CollisionShape2D::_get_ancestor_collision_object() const {
+	Node *parent = get_parent();
+	while (parent) {
+		CollisionObject2D *co = Object::cast_to<CollisionObject2D>(parent);
+		if (co) {
+			return co;
+		}
+		parent = parent->get_parent();
+	}
+	return nullptr;
+}
+
+Transform2D CollisionShape2D::_get_transform_to_collision_object() const {
+	Transform2D transform_to_col_obj = get_transform();
+	Node *parent = get_parent();
+	while (parent != collision_object) {
+		CanvasItem *parent_2d = Object::cast_to<CanvasItem>(parent);
+		transform_to_col_obj = parent_2d->get_transform() * transform_to_col_obj;
```

Comment:
cast_to can return null, such as if a non-CanvasItem is nested in between.

Reply:
Thanks, added a check.